### PR TITLE
Preserve removed navigation PropertyInfo

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

**Comment by lajones** [comment]
The change looks good but could you add something to say why you are doing this - what problem does it solve? And can you add a test?

File: src/Microsoft.EntityFrameworkCore/Metadata/Internal/ForeignKey.cs [hunk 1]
```diff
@@ -213,14 +213,16 @@ public virtual void UpdatePrincipalToDependentConfigurationSource(ConfigurationS
                         DeclaringEntityType.Model.ConventionDispatcher.OnNavigationRemoved(
                             DeclaringEntityType.Builder,
                             PrincipalEntityType.Builder,
-                            oldNavigation.Name);
+                            oldNavigation.Name,
+                        	oldNavigation.PropertyInfo);
```
Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore/Metadata/Internal/ForeignKey.cs]: tabbing

**Comment by smitpatel** [comment]
@lajones - From what I understand, this change stops `RelationalshipDiscoveryConvention` to run while a shadow navigation is removed because convention would not find anything new on CLR type.

**Comment by AndriySvyryd** [comment]
@lajones @smitpatel is correct. This is a perf fix. By itself it doesn't improve much, but it's a prerequisite for #5358

**Pull request merged**